core/api/dev/apollo-federation/supergraph.graphql

```diff
@@ -545,6 +550,20 @@ input IntraLedgerUsdPaymentSendInput
   walletId: WalletId!
 }
 
+"""A lightning invoice."""
+interface Invoice
+  @join__type(graph: PUBLIC)
+{
+  """The payment hash of the lightning invoice."""
+  paymentHash: PaymentHash!
+
+  """The bolt11 invoice to be paid."""
+  paymentRequest: LnPaymentRequest!
+
+  """The payment secret of the lightning invoice."""
```

Comment:
There can be confusion between secret and preimage. Is this really the secret here? We are not exposing the image? We should/could after the payment has completed

Reply:
The [domain type](https://github.com/GaloyMoney/galoy/blob/f052151cdd4ec01b7e418c410be284e89b8df275/core/api/src/domain/bitcoin/lightning/index.types.d.ts#L110) of that field is `PaymentIdetifyingSecret` which I dont think is the preimage, but I could be mistaken. It is confusing because we call the preimage the `secret` on the [WalletInvoice type](https://github.com/GaloyMoney/galoy/blob/f052151cdd4ec01b7e418c410be284e89b8df275/core/api/src/domain/wallet-invoices/index.types.d.ts#L87) . For this pr, I haven't changed the value we are exposing for this field.

Reply:
At least we can clarify with the description

Reply:
I think the preimage is actually not that useful for the invoice/receive side, but it's important on the sender side, because payment request (signed by the recipient) + preimage = proof of payment